cli/src/config.rs

```diff
@@ -412,6 +423,7 @@ struct _Config {
     provider: Provider,
     workspace: Option<WorkspaceConfig>,
     scripts: Option<ScriptsConfig>,
+    hooks: Option<HooksConfig>,
```

Comment:
you can make this a `Hashmap/BTreeMap` and then fail during conversion from `_Config` to `Config` if you encounter an unknown hook

Reply:
@paul-schaaf I actually started with this and switched to a struct since we know in advance the hooks we're supporting, compared to `scripts` which is arbitrary. I can switch to `Hashmap/BTreeMap` if you feel rejecting invalid ones is more valuable though, yes.

Reply:
I feel like it's not too ugly implementation-wise and provides a better UX.

and just to be clear what I mean is: `Hashmap/BTreeMap` in `_Config` and `Option<HooksConfig>` in `Config`

Reply:
Ok got it. WIll give that a go tmr. Eager to get that merged :smile:

Reply:
Alright provided unsupported hooks now results in an error:
```sh
thread 'main' panicked at 'Error reading hooks. Hooks ["WRONG"] are not part of valid hooks ["pre_build", "post_build", "pre_test", "post_test", "pre_deploy", "post_deploy"].', cli/src/config.rs:336:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Lmk if you any other remarks.

Reply:
@paul-schaaf bump, in case this fell through the cracks